activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -88,7 +88,7 @@ def find_by!(arg, *args)
       where(arg, *args).take!
     rescue ::RangeError
       raise RecordNotFound.new("Couldn't find #{@klass.name} with an out of range value",
-                               @klass.name)
+                               @klass.name, @klass.primary_key)
```

Comment:
Should we pass `args` as well?

Reply:
@prathamesh-sonpatki fourth argument is for ids. Here, however, args can be anything. We can of course try to extract ids from arguments, but i don't think it's worth it tbh.